### PR TITLE
Refactor processing funcs with tests

### PR DESCRIPTION
**Pull request by gkallenberg**

### CODE REVIEW

File: src/helpers/RecordsProcessor.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  })
+  logger.info(`Processing completed.  Successful transformations -  ${success}.  Failed transformations - ${failure}.`)
+  return output
+}
+
+const decodeAvro = function (type, record) {
+  let decodedRecord = Buffer.from(record, 'base64')
+  try {
+    return type.fromBuffer(decodedRecord)
+  } catch (e) {
```
Comment (rhernand3z) [inline src/helpers/RecordsProcessor.js:34]: would be a good place to log the error before returning false.

File: src/helpers/RecordsProcessor.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import avro from 'avsc'
+import logger from './Logger'
+
+// Map records to decode Avro and return JSON as data to firehose.
+const processRecords = function (schema, records) {
+  let success = 0
+  let failure = 0
+  let type = avro.Type.forSchema(schema)
+  const output = records.map((record) => {
```
Comment (EdwinGuzman) [inline src/helpers/RecordsProcessor.js:9]: You'll run into problems if `records` is not an array and you try to run `.map` on it. You should check to see if `records` is an array first.

File: src/helpers/RecordsProcessor.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    return type.fromBuffer(decodedRecord)
+  } catch (e) {
+    logger.error(`Decoding fatal error occurred: ${e.message}`, { debugInfo: e })
+    return false
+  }
+}
+
+export {
+  processRecords,
+  decodeAvro
```
Comment (EdwinGuzman) [inline src/helpers/RecordsProcessor.js:42]: You don't need to export `decodeAvro` if you're just using it in this file.

**Comment by EdwinGuzman** [comment]
@gkallenberg looks good! Just minor comments.

**Pull request merged**